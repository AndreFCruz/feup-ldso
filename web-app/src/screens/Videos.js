import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { setFaculty } from '../actions/facultyActions';
import { getVideos, editVideo } from '../actions/videosActions';
import PromotionalVideo from '../components/PromotionalVideo';

class Videos extends Component {

    componentDidMount() {
        this.updateFaculty();
    }

    componentDidUpdate(prevProps) {
        const { faculty } = this.props;

        if (prevProps.faculty !== faculty)
            this.updateFaculty();
    }

    updateFaculty() {
        const { match, faculty, getVideos, setFaculty } = this.props;

        setFaculty(match.params.faculty);
        getVideos(faculty, 'en');
        getVideos(faculty, 'pt');
    }

    render() {
        const { loading, loadingAction, faculty, videosEN, videosPT, editVideo } = this.props;

        return (
            <div style={contentStyle}>
                <PromotionalVideo loading={loading} loadingAction={loadingAction} faculty={faculty} videosEN={videosEN} videosPT={videosPT} mainAction={editVideo}/>
            </div>
        );
    }
}

const contentStyle = {
    display: 'flex',
    flexDirection: 'row'
};

Videos.propTypes = {
    loading: PropTypes.bool,
    loadingAction: PropTypes.bool,
    faculty: PropTypes.string,
    videosEN: PropTypes.object,
    videosPT: PropTypes.object,
    setFaculty: PropTypes.func,
    getVideos: PropTypes.func,
    editVideo: PropTypes.func,
    match: PropTypes.object
};

const mapStateToProps = ({ faculty, videos }) => ({
    faculty: faculty.name,
    loading: videos.loading,
    loadingAction: videos.loadingAction,
    videosEN: videos.videosEN,
    videosPT: videos.videosPT
});


const mapDispatchToProps = {
    setFaculty,
    getVideos,
    editVideo
};

export default connect(mapStateToProps, mapDispatchToProps)(Videos);
