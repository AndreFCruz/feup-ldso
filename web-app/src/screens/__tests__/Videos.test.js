import React from 'react';
import Videos from '../Videos';
import Enzyme, { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

const middlewares = []; // you can mock any middlewares here if necessary
const mockStore = configureStore(middlewares);

function setup(videos) {

    const initialState = {
        faculty: {
            name: 'faup'
        },
        videos: {
            loading: false,
            loadingAction: false,
            videosEN: videos,
            videosPT: videos,
        }
    };

    const props = {
        match: {
            params: {
                faculty: 'faup',
            }
        },
        setFaculty: jest.fn(),
        editVideo: jest.fn(),
        getVideos: jest.fn(),
    };

    const wrapper = shallow(<Videos {...props} />,
        {
            context: { store: mockStore(initialState) },
            disableLifecycleMethods: true
        });

    return wrapper;
}

it('renders loading screen', () => {

    const videos = [];

    const wrapper = setup(videos);

    expect(wrapper.dive()).toMatchSnapshot();
});
