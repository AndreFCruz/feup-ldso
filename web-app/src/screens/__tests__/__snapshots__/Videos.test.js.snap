// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot renders without crashing 1`] = `
<div>
  <div
    className="intro"
  >
    <img
      alt="application logo"
      src="test-file-stub"
      width="140"
    />
  </div>
  <div>
    <div
      className="navigator"
    >
      <h6
        className="fadeup"
      >
        <a
          aria-current={null}
          href="/faculties/fadeup"
          onClick={[Function]}
        >
          FADEUP
        </a>
      </h6>
      <h6
        className="faup"
      >
        <a
          aria-current={null}
          href="/faculties/faup"
          onClick={[Function]}
        >
          FAUP
        </a>
      </h6>
      <h6
        className="fbaup"
      >
        <a
          aria-current={null}
          href="/faculties/fbaup"
          onClick={[Function]}
        >
          FBAUP
        </a>
      </h6>
      <h6
        className="fcnaup"
      >
        <a
          aria-current={null}
          href="/faculties/fcnaup"
          onClick={[Function]}
        >
          FCNAUP
        </a>
      </h6>
      <h6
        className="fcup"
      >
        <a
          aria-current={null}
          href="/faculties/fcup"
          onClick={[Function]}
        >
          FCUP
        </a>
      </h6>
      <h6
        className="fdup"
      >
        <a
          aria-current={null}
          href="/faculties/fdup"
          onClick={[Function]}
        >
          FDUP
        </a>
      </h6>
      <h6
        className="fep"
      >
        <a
          aria-current={null}
          href="/faculties/fep"
          onClick={[Function]}
        >
          FEP
        </a>
      </h6>
      <h6
        className="feup"
      >
        <a
          aria-current={null}
          href="/faculties/feup"
          onClick={[Function]}
        >
          FEUP
        </a>
      </h6>
      <h6
        className="ffup"
      >
        <a
          aria-current={null}
          href="/faculties/ffup"
          onClick={[Function]}
        >
          FFUP
        </a>
      </h6>
      <h6
        className="flup"
      >
        <a
          aria-current={null}
          href="/faculties/flup"
          onClick={[Function]}
        >
          FLUP
        </a>
      </h6>
      <h6
        className="fmdup"
      >
        <a
          aria-current={null}
          href="/faculties/fmdup"
          onClick={[Function]}
        >
          FMDUP
        </a>
      </h6>
      <h6
        className="fmup"
      >
        <a
          aria-current={null}
          href="/faculties/fmup"
          onClick={[Function]}
        >
          FMUP
        </a>
      </h6>
      <h6
        className="fpceup"
      >
        <a
          aria-current={null}
          href="/faculties/fpceup"
          onClick={[Function]}
        >
          FPCEUP
        </a>
      </h6>
      <h6
        className="icbas"
      >
        <a
          aria-current={null}
          href="/faculties/icbas"
          onClick={[Function]}
        >
          ICBAS
        </a>
      </h6>
    </div>
    <div
      className="container null"
    >
      <div>
        <h1>
          Redirect to App login or another admin page
        </h1>
      </div>
    </div>
  </div>
  <div
    className="notification-container notification-container-empty"
  >
    <span />
  </div>
</div>
`;

exports[`renders loading screen 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Videos
    editVideo={[Function]}
    faculty="faup"
    getVideos={[Function]}
    loading={false}
    loadingAction={false}
    match={
      Object {
        "params": Object {
          "faculty": "faup",
        },
      }
    }
    setFaculty={[Function]}
    videosEN={Array []}
    videosPT={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <withRouter(PromotionalVideo)
        faculty="faup"
        loading={false}
        loadingAction={false}
        mainAction={[Function]}
        videosEN={Array []}
        videosPT={Array []}
      />,
      "style": Object {
        "display": "flex",
        "flexDirection": "row",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "faculty": "faup",
        "loading": false,
        "loadingAction": false,
        "mainAction": [Function],
        "videosEN": Array [],
        "videosPT": Array [],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <withRouter(PromotionalVideo)
          faculty="faup"
          loading={false}
          loadingAction={false}
          mainAction={[Function]}
          videosEN={Array []}
          videosPT={Array []}
        />,
        "style": Object {
          "display": "flex",
          "flexDirection": "row",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "faculty": "faup",
          "loading": false,
          "loadingAction": false,
          "mainAction": [Function],
          "videosEN": Array [],
          "videosPT": Array [],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
